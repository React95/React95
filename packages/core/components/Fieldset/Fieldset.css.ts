import { style } from '@vanilla-extract/css';
import { contract } from '../themes/contract.css';

export const field = style({
  borderTopStyle: 'groove',
  borderTopWidth: contract.space[2],
  borderTopColor: contract.colors.borderLighter,
  borderLeftStyle: 'groove',
  borderLeftWidth: contract.space[2],
  borderLeftColor: contract.colors.borderLighter,
  borderBottomStyle: 'solid',
  borderBottomWidth: contract.space[1],
  borderBottomColor: contract.colors.borderDark,
  borderRightStyle: 'solid',
  borderRightWidth: contract.space[1],
  borderRightColor: contract.colors.borderDark,
  boxShadow: `1px 1px 0 0 ${contract.colors.borderLighter}`,
});

export const legend = style({
  paddingLeft: contract.space[4],
  paddingRight: contract.space[1],
});
