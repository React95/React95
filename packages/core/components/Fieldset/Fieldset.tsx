import React, { forwardRef } from 'react';
import * as styles from './Fieldset.css';
import { Frame, FrameProps } from '../Frame/Frame';

export type FieldSetProps = {
  legend?: string;
} & React.HTMLProps<HTMLFieldSetElement> &
  FrameProps;

export const Fieldset = forwardRef<HTMLFieldSetElement, FieldSetProps>(
  ({ legend, children, ...rest }, ref) => (
    <Frame as="fieldset" {...rest} ref={ref} className={styles.field}>
      {legend && <legend className={styles.legend}>{legend}</legend>}
      {children}
    </Frame>
  ),
);
