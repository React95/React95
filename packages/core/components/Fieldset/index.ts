import { Fieldset } from './Fieldset';

export default Fieldset;
