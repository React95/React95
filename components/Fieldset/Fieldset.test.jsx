import React from 'react';
import Fieldset from './Fieldset';
import { render } from 'react-testing-library';

describe('<Fieldset />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with legend prop and children element', () => {
      const { container } = render(
        <Fieldset legend="example">
          <p>something</p>
        </Fieldset>
      );
      expect(container).toMatchSnapshot();
    });
  });
});
