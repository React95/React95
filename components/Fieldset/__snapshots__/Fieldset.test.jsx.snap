// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fieldset /> Snapshots should match snapshot with legend prop and children element 1`] = `
.c0 {
  border-top-style: groove;
  border-top-width: 2px;
  border-top-color: #e6e6e6;
  border-left-style: groove;
  border-left-width: 2px;
  border-left-color: #e6e6e6;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: #848284;
  border-right-style: solid;
  border-right-width: 1px;
  border-right-color: #848284;
  box-shadow: 1px 1px 0 0 #e6e6e6;
}

.c1 {
  padding-left: 4px;
  padding-right: 1px;
}

<div>
  <fieldset
    class="c0"
  >
    <legend
      class="c1"
    >
      example
    </legend>
    <p>
      something
    </p>
  </fieldset>
</div>
`;
